beamer/health/notification_state.py

```diff
@@ -0,0 +1,34 @@
+import json
+import os
+from pathlib import Path
+
+
+class NotificationState:
+    cache_path: Path
+
+    def __init__(self, cache_path: Path) -> None:
+        self.cache_path = cache_path
+
+        try:
+            with open(os.path.join(self.cache_path, "breadcrumbs.json"), "r") as f:
+                self.notified_state = json.load(f)
+        except FileNotFoundError:
+            self.notified_state = {}
+
+    def is_set(self, request_id: str, notified_on: str) -> bool:
+        breadcrumb = self.notified_state.get(str(request_id))
+        if breadcrumb is None:
+            return False
+        else:
+            return breadcrumb.get(notified_on, False)
```

Comment:
Nitpick: could be more concisely written:
```python
return breadcrumb is not None and breadcrumb.get(notified_on, False)
```

Reply:
yes, but what is concisely for you is basically unreadable for me, because I don't live in python :)

Reply:
How about:
```
        if breadcrumb is None:
            return False
        
        return breadcrumb.get(notified_on, False)
```
?

Reply:
Perfect. :+1: